rolling/ci-drake-ros-underlay.yaml

```diff
@@ -0,0 +1,64 @@
+%YAML 1.1
+# ROS buildfarm ci-build file
+---
+build_environment_variables:
+  ROS_PYTHON_VERSION: '3'
+build_tool: colcon
+build_tool_args: '--merge-install --cmake-args -DCMAKE_BUILD_TYPE=Release -DSKIP_MULTI_RMW_TESTS=1 --no-warn-unused-cli'
+build_tool_test_args: '--merge-install --retest-until-pass 2 --ctest-args -LE "(linter|performance|xfail| placeholder-with-space )" --pytest-args -m "not linter and not performance and not xfail"'
+install_packages:
+- python3-colcon-bash
+jenkins_job_label: ci-agent
+jenkins_job_priority: 50
+jenkins_job_schedule: 15 23 * * 6
+jenkins_job_timeout: 300
+jenkins_job_weight: 4
+package_dependencies: true
+package_names:
+- ament_cmake_clang_format
+- cv_bridge
+- ros_base
+- rviz2
+package_selection_args: '--packages-ignore rmw_fastrtps_dynamic_cpp --packages-ignore-regex .*connext.* .*cyclonedds.* .*iceoryx.*'
```

Comment:
I'm just curious: why do we have both `build_tool_args` and `package_selection_args`?  Couldn't we do this all as part of `build_tool_args`?

Reply:
This is part of the mechanism that the CI jobs use to ensure minimal dependencies. Any packages present in the workspace that aren't found by the `package_selection_args` are marked with an ignore file. The dependencies are enumerated after that, then build, then test.